src/datastore/pg-write-store.ts

```diff
@@ -1963,13 +1965,33 @@ export class PgWriteStore extends PgStore {
     for (const txId of txIds) {
       logger.verbose(`Restoring mempool tx: ${txId}`);
     }
-    const updateResults = await sql<{ tx_id: string }[]>`
+
+    const updatedRows = await sql<{ tx_id: string }[]>`
       UPDATE mempool_txs
       SET pruned = false
       WHERE tx_id IN ${sql(txIds)}
       RETURNING tx_id
     `;
-    const restoredTxs = updateResults.map(r => r.tx_id);
+
+    const restoredTxs = updatedRows.map(r => r.tx_id);
+
+    // txs that didnt exist in the mempool need to be inserted into the mempool
+    if (updatedRows.length < txIds.length) {
+      const updatedTxs = updatedRows.map(r => r.tx_id);
+      const txsRequiringInsertion = txIds.filter(txId => !updatedTxs.includes(txId));
+
+      const txs: TxQueryResult[] = await sql`
+        SELECT ${sql(TX_COLUMNS)} FROM txs
+        WHERE tx_id IN ${sql(txsRequiringInsertion)}
+      `;
+
+      const mempoolTxs = convertTxQueryResultToDbMempoolTx(txs);
+
+      await this.updateMempoolTxs({ mempoolTxs });
+
+      restoredTxs.concat(txsRequiringInsertion);
```

Comment:
The `concat` doesn't modify an existing array, a new one is created but isn't used here